OpenRA.Mods.Common/Traits/BotModules/HarvesterBotModule.cs

```diff
@@ -170,6 +184,33 @@ void IBotTick.BotTick(IBot bot)
 			}
 		}
 
+		void SellRefineryTooClose(IBot bot)
+		{
+			var refineries = world.ActorsHavingTrait<Refinery>().Where(a => a.Owner == player).ToArray();
+			var found = false;
+			int i = 0, j = 0;
+
+			while (i < refineries.Length)
+			{
+				if (found)
+					break;
+				j = i + 1;
+				while (j < refineries.Length)
+				{
+					if ((refineries[i].Location - refineries[j].Location).LengthSquared <= Info.SellRefineryCloseCellDistance * Info.SellRefineryCloseCellDistance)
+					{
+						bot.QueueOrder(new Order("Sell", refineries[i], Target.FromActor(refineries[i]), false));
+						found = true;
+						break;
+					}
+
+					j++;
+				}
+
+				i++;
+			}
```

Comment:
```suggestion
			for (var i = 0; i < refineries.Length; i++)
			{
				for (var j = i + 1; j < refineries.Length; j++)
				{
					if ((refineries[i].Location - refineries[j].Location).LengthSquared <= Info.SellRefineryCloseCellDistance * Info.SellRefineryCloseCellDistance)
					{
						bot.QueueOrder(new Order("Sell", refineries[i], Target.FromActor(refineries[i]), false));
						return;
					}
				}
			}
```

Just use for loops. Also shouldn't we sell refinery that's farthest away from resources?

Reply:
Well, you can just take "additional" commit for better test of the first three commit. those will make AI survive longer in AI test and build refinery to expasion base more often. 

> Also shouldn't we sell refinery that's farthest away from resources?

there is a problem is "BaseBuilder" we have now always build refinery at somewhere far from resource, really annoying (I think "BaseBuilder" may also need some change on place structure to help refinery get to resource). I tried and gave up as soon as I had enough of "BaseBuilder".